[Bug] @dispatch over @torch.compiler.disable is silently discarded — top-level backend dispatch never fires

### Describe the bug

For ops decorated as

```python
@dispatch('kda')
@torch.compiler.disable
def chunk_kda(...):
```

the dispatch wrapper is silently discarded at import time: backend selection never runs for these entry points, and calls always fall through to the default implementation. This affects at least:

- `fla/ops/kda/chunk.py` — `chunk_kda`
- `fla/ops/kda/gate.py:331` — `fused_kda_gate`
- `fla/ops/gated_delta_rule/chunk.py:395` — `chunk_gated_delta_rule`

A consequence: the `flash_kda` backend merged in #852 is never actually selected through `chunk_kda`. Its tests (`test_flash_kda_chunk*`) still pass because they compare against a fp64 gold reference, which the default Triton path also matches — so the backend being dead is invisible to CI.

**Root cause**: `torch.compiler.disable` (torch 2.8, `torch._dynamo.disable`) unwraps `fn = innermost_fn(fn)`, following `_torchdynamo_orig_callable`. The evaluation order is:

1. `D1 = torch.compiler.disable(raw)` — D1 carries `_torchdynamo_orig_callable = raw` in its `__dict__`.
2. `dispatch(...)` wraps D1 with `@functools.wraps(D1)` — which **copies D1's `__dict__`, including `_torchdynamo_orig_callable`**, onto the dispatch wrapper.
3. The dispatch decorator finally re-applies `torch.compiler.disable(wrapper)` — `innermost_fn(wrapper)` follows the copied `_torchdynamo_orig_callable` straight to `raw` and wraps *that* instead.

The returned object is `DisableContext(raw)`; the dispatch wrapper is dropped entirely.

I also verified at runtime that calling `chunk_kda(...)` never reaches the dispatch loop (instrumented `fla/ops/backends/__init__.py`), while sub-ops with a bare `@dispatch` (e.g. `chunk_kda_fwd_intra`) dispatch fine.

**Suggested fix**: swap the decorator order so `dispatch` wraps the raw function and `compiler.disable` is outermost:

```python
@torch.compiler.disable
@dispatch('kda')
def chunk_kda(...):
```

With this order, `functools.wraps` copies the *raw* function's `__dict__` (no `_torchdynamo_orig_callable`), `innermost_fn` keeps the dispatch wrapper, dispatch fires, and the whole call is still excluded from compile graphs by the outer `disable`. Verified locally: after swapping, `chunk_kda` calls route through backend verification/selection as intended, and the full `tests/ops/test_kda.py` suite passes (with the `flash_kda` backend now genuinely exercised in inference mode).

One behavior note: fixing this *activates* the `flash_kda` backend for inference-mode calls (it has `default_enable=True`), which is the intent of #852 but is an observable behavior change worth calling out in the changelog.

### Versions

- fla `main` as of 2026-08-07 (61321044) — decorator order unchanged
- torch **2.8.0+cu128**, triton **3.4.0**
- GPU: RTX 5090 (sm_120a), CUDA 12.8

### Steps to reproduce

Minimal repro of the decorator interaction (no fla import needed):

```python
import functools, torch

calls = []

def my_dispatch(fn):
    @functools.wraps(fn)
    def wrapper(*a, **k):
        calls.append("dispatched")
        return fn(*a, **k)
    return torch.compiler.disable(wrapper)

@my_dispatch
@torch.compiler.disable
def f():
    return "ok"

f()
assert calls == ["dispatched"], f"dispatch wrapper discarded: {calls}"
# -> AssertionError: dispatch wrapper discarded: []
```

Against fla itself: set `FLA_FLASH_KDA=1`, call `chunk_kda` under `torch.inference_mode()` with a supported config, and instrument `fla/ops/backends/__init__.py` (or spy on the backend's `chunk_kda`) — the dispatch loop never runs and the Triton path always executes. After the decorator swap, the same call routes to the `flash_kda` backend.

--- a/fla/ops/gated_delta_rule/chunk.py
+++ b/fla/ops/gated_delta_rule/chunk.py
@@ -392,8 +392,8 @@ class ChunkGatedDeltaRuleFunction(torch.autograd.Function):
         )
 
 
-@dispatch('gated_delta_rule')
 @torch.compiler.disable
+@dispatch('gated_delta_rule')
 def chunk_gated_delta_rule(
     q: torch.Tensor,
     k: torch.Tensor,

--- a/fla/ops/kda/chunk.py
+++ b/fla/ops/kda/chunk.py
@@ -173,8 +173,8 @@ class ChunkKDAFunction(torch.autograd.Function):
                 None, None, None, None, None, None, None, None, None, None, None, None, None, None)
 
 
-@dispatch('kda')
 @torch.compiler.disable
+@dispatch('kda')
 def chunk_kda(
     q: torch.Tensor,
     k: torch.Tensor,

--- a/fla/ops/kda/gate.py
+++ b/fla/ops/kda/gate.py
@@ -328,8 +328,8 @@ class KDAGateFunction(torch.autograd.Function):
         return dg, dA, dbias, None, None
 
 
-@dispatch('kda')
 @torch.compiler.disable
+@dispatch('kda')
 def fused_kda_gate(
     g: torch.Tensor,
     A_log: torch.Tensor,

